src/prompt/cursor.rs

```diff
@@ -2,10 +2,27 @@ use std::fmt::{Display, Formatter, Result};
 
 use zeroize::ZeroizeOnDrop;
 
-#[derive(Default, ZeroizeOnDrop, Clone)]
+#[derive(ZeroizeOnDrop, Clone)]
+#[cfg_attr(not(feature = "multiline"), derive(Default))]
 pub struct StringCursor {
     value: Vec<char>,
     cursor: usize,
+    #[cfg(feature = "multiline")]
```

Comment:
@kingwingfly  Currently I'm thinking about how could your patch be improved with more decoupling from the main source code and simplification.
First thing about the cursor: I believe `editing` is not the property of the cursor, it's property of the input prompt (UI). And `multiline` could be "always enabled" cause it's about the `\n` in the string: it is inserted from the outside or not. Removing these two fields (moving them outside of the cursor) will make it simpler and context-independent. Also, less of cfg switches.

Reply:
I tried your suggestion.

First, I remove the two flags from `StringCursor`, and add `editing` to `Input`. This leads that I cannot switch `editing` flag correctly and decide whether to render footer without `multiline` flag.  And `Tab` will always switch editing flag on one-line input. 

Then I add `multiline` flag to `Input`. But `PromptInteraction::interact_on_prepared` cannot got `self.multiline` because it's in trait. It will always insert `\n`. 

Then I move this job to `PromptInteraction::on`. It's used to decide what's the next State, but I modify input there, that's strange. 

Finally, I think `multiline` and `editing` should be the attributes of a `StringCursor`. `multiline` stands for whether it accept '\n'. `editing` stands for whether it's mutable. 

I'd like to make these two flags public instead...

Reply:
> multiline could be "always enabled"

I misunderstood this before. (I used to think this asks to remove flag in StringCursor)

I thought keeping `multiline` as feature may be better since it add two byte to Input, and many may not want multiline at all. 

I'm not a CS student and don't have any practical experience. This may be quite performance anxiety, if you insist to make it always enable and remove all `cfg feature`, I'll try tonight.

Reply:
@kingwingfly your code doesn't degrade performance, it only contributes to the binary size... so, we might even consider to remove the feature flag. Anyways, I need to take time to play with your patch to not load you with pointless tasks.

Reply:
@kingwingfly if you have capacity,  could you check how much the binary size of a basic example increases with the multiline feature enabled and how longer is the build time?

Reply:
Okey, I believe there's little influence on performance or binary size. 

But here are two facts as so far:
1. If feature removed, always enable, for those who don't need `multiline`, there would be two useless flags, `multiline` and `editing`.
2. The two flag can be in `Input` or `StringCursor`, and both ways work. However, the frontier needs modify `Input.input` in `PromptInteraction::on`. Although other prompts like `Select` changes cursor in `on`, I prefer gathering input-modifying in `PromptInteraction::interact_on_prepared` as before, and `on` only for state transfer.

So, there are two question left:
1. Whether enable `multiline` by default and remove the feature flag or not?
2. `multiline` and `editing` flags should belong to `Input` or `StringCursor`?